include/alpaka/atomic/AtomicUniformCudaHipBuiltIn.hpp

```diff
@@ -177,22 +177,51 @@ namespace alpaka
         {
             //
             __device__ static auto atomicOp(
-                AtomicUniformCudaHipBuiltIn const&,
+                AtomicUniformCudaHipBuiltIn const& ctx,
                 [[maybe_unused]] unsigned long int* const addr,
                 [[maybe_unused]] unsigned long int const& value) -> unsigned long int
             {
-                // LLP64
-                if constexpr(std::numeric_limits<unsigned int>::max() == std::numeric_limits<unsigned long>::max())
-                    return ::atomicSub(reinterpret_cast<unsigned int*>(addr), static_cast<unsigned int>(value));
```

Comment:
I don't think we should remove the `LLP64` case when it is available natively.

Reply:
So just calling `alpaka::atomicAdd` for the non `LLP64` case and leave `atomicSub` for `LLP64`?

Reply:
Yes, that is better IMO.